modules/auth/proc-adding-gitlab-oauth-as-an-authentication-provider.adoc

```diff
@@ -0,0 +1,103 @@
+[id='proc-adding-gitlab-oauth-as-an-authentication-provider_{context}']
+= Adding GitLab OAuth as an authentication provider in {product}
+
+The {product} `core-plugin-api` package includes a GitLab authentication provider that can authenticate users by using GitLab OAuth.
+
+.Procedure
+. To allow {product} to authenticate with Gitlab, create an OAuth Application in Gitlab.
++
+Go to https://gitlab.com/-/profile/applications[GitLab User settings > Applications], and click the *Add new application* button.
+
+*Name*:: Enter your application name, such as `{product}`.
+*Redirect URI*:: Enter the backend authentication URI set in {product}:
+For local development::: `pass:[http://localhost:7007/api/auth/gitlab/handler/frame]`
+For a non-local deployment::: `pass:[http://{APP_FQDN}:{APP_BACKEND_PORT}/api/auth/gitlab/handler/frame]`
```

Comment:
```suggestion
For a non-local deployment::: `pass:[http://{APP_FQDN}/api/auth/gitlab/handler/frame]`
```
it should not point to backend port for rhdhd deployed in a production environment. It should point to frontend and that one is usually default one (https or HTTP) so we don't need to specify it here